### PR TITLE
docs: add info about caching and describe useVisitorData hook

### PR DESCRIPTION
**Pull request by ilfa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,9 +200,18 @@ Custom prefix for localStorage and sessionStorage cache keys. Will be ignored if
...
 `useVisitorData` hook performs identification requests with the FingerprintJS Pro API. The returned object contains information about loading status, errors, and [visitor](https://dev.fingerprintjs.com/docs/js-agent#extendedresultac).
+
+#### Params
 - `getOptions: GetOptions<TExtended>` parameter follows parameters of the FingerprintJS Pro's [`get` function](https://dev.fingerprintjs.com/docs/js-agent#parameters-reference).
 - `config: UseVisitorDataConfig`'s property `immediate` determines whether the `getData()` method will be called immediately after the hook mounts or not.
 
+#### Returned object
+- `getData: (ignoreCache: boolean) => Promise<VisitorData>` Performs identification request to server and returns visitors data. When it is called, a FingerprintJS server will send a [webhook request](https://dev.fingerprintjs.com/docs/webhooks) to your server.
+- `isLoading: boolean` Indicates `getData` request status
+- `data: VisitorData` Contains visitors data requested after after `getData()` call
```
Comment (TheUnderScorer) [inline README.md:211]: You have duplicate "after" here 😁

File: README.md [hunk 2]
```diff
@@ -164,6 +165,13 @@ export default App;
 
 See the full code example in the [examples folder](https://github.com/fingerprintjs/fingerprintjs-pro-react/tree/main/examples/spa).
 
+## Caching strategy
+:warning: **WARNING** If you use data from `extendedResult`, please pay additional attention to caching strategy
```
Comment (makma) [inline README.md:169]:
```suggestion
:warning: **WARNING** If you use data from `extendedResult`, please pay additional attention to caching strategy.
```

File: README.md [hunk 3]
```diff
@@ -164,6 +165,13 @@ export default App;
...
+
+FingerprintJS Pro uses API calls as the basis for billing. Our [best practices](https://dev.fingerprintjs.com/docs/caching-visitor-information) strongly recommend using cache for saving your money. Library use `SessionStorage` cache strategy by default.
```
Comment (makma) [inline README.md:171]:
```suggestion
FingerprintJS Pro uses API calls as the basis for billing. Our [best practices](https://dev.fingerprintjs.com/docs/caching-visitor-information) strongly recommend using cache to optimise API calls rate. The Library uses the SessionStorage cache strategy by default.
```

File: README.md [hunk 4]
```diff
@@ -164,6 +165,13 @@ export default App;
...
+
+But if you need some [`extendedResult`](https://dev.fingerprintjs.com/docs/js-agent#extendedresult) fields that can change, like `ip` or `lastSeenAt` it is better to pass `ignoreCache=true` inside [`getData`](#returned-object) function.
```
Comment (makma) [inline README.md:173]:
```suggestion
Some fields from the [extendedResult](https://dev.fingerprintjs.com/docs/js-agent#extendedresult) (e.g `ip` or `lastSeenAt`) might change for the same visitor. If you need exact current data, it is recommended to pass `ignoreCache=true` inside [getData](#returned-object) function.
```

File: README.md [hunk 5]
```diff
@@ -193,9 +201,18 @@ Custom prefix for localStorage and sessionStorage cache keys. Will be ignored if
 `useVisitorData(getOptions, config)`
 
 `useVisitorData` hook performs identification requests with the FingerprintJS Pro API. The returned object contains information about loading status, errors, and [visitor](https://dev.fingerprintjs.com/docs/js-agent#extendedresultac).
+
+#### Params
 - `getOptions: GetOptions<TExtended>` parameter follows parameters of the FingerprintJS Pro's [`get` function](https://dev.fingerprintjs.com/docs/js-agent#parameters-reference).
 - `config: UseVisitorDataConfig`'s property `immediate` determines whether the `getData()` method will be called immediately after the hook mounts or not.
 
+#### Returned object
+- `getData: (ignoreCache: boolean) => Promise<VisitorData>` Performs identification request to server and returns visitors data. When it is called, a FingerprintJS server will send a [webhook request](https://dev.fingerprintjs.com/docs/webhooks) to your server.
```
Comment (makma) [inline README.md:210]: I'd remove the sentence about webhooks.

File: README.md [hunk 6]
```diff
@@ -193,9 +201,18 @@ Custom prefix for localStorage and sessionStorage cache keys. Will be ignored if
...
+- `isLoading: boolean` Indicates `getData` request status
```
Comment (makma) [inline README.md:211]:
```suggestion
- `isLoading: boolean` Indicates `getData` request status.
```

File: README.md [hunk 7]
```diff
@@ -193,9 +201,18 @@ Custom prefix for localStorage and sessionStorage cache keys. Will be ignored if
...
+- `data: VisitorData` Contains visitors data requested after after `getData()` call
```
Comment (makma) [inline README.md:212]:
```suggestion
- `data: VisitorData` Contains visitors data requested after `getData()` call.
```

File: README.md [hunk 8]
```diff
@@ -193,9 +201,18 @@ Custom prefix for localStorage and sessionStorage cache keys. Will be ignored if
...
+- `error: Error` Error information in case of failed request
```
Comment (makma) [inline README.md:213]:
```suggestion
- `error: Error` Error information in case the request failed.
```

**Pull request merged**